Common/Source/Dialogs/dlgRadioSettings.cpp

```diff
@@ -154,48 +157,56 @@ int Idx=0;
   {
     RadioPara.Changed =FALSE;
     TCHAR Name[250];
-    if(_tcscmp(_T("        "), RadioPara.ActiveName ) == 0)
+    if((_tcscmp(_T("        "), RadioPara.ActiveName ) == 0)  || ( _tcslen(RadioPara.ActiveName) == 0))
       Idx = SearchStation(RadioPara.ActiveFrequency);
     if(Idx !=0)
     {
-        _stprintf(RadioPara.ActiveName,_T("%s"),WayPointList[Idx].Name);
+
+    		CopyTruncateString(RadioPara.ActiveName, NAME_SIZE, WayPointList[Idx].Name);
         ActiveRadioIndex = Idx;
         if( HoldOff ==0)
         {
           HoldOff = HOLDOFF_TIME;
           devPutFreqActive(RadioPara.ActiveFrequency, WayPointList[Idx].Name);
         }
     }
+    TCHAR ActiveName[DEVICE_NAME_LEN+8];
+		CopyTruncateString(ActiveName, DEVICE_NAME_LEN, RadioPara.ActiveName);
+
     if(RadioPara.TX)
-      _stprintf(Name,_T(">%s<"),RadioPara.ActiveName);
+      _stprintf(Name,_T(">%s<"),ActiveName);
     else
       if(RadioPara.RX_active)
-        _stprintf(Name,_T("<%s>"),RadioPara.ActiveName);
+        _stprintf(Name,_T("<%s>"),ActiveName);
       else
-        _stprintf(Name,_T("[%s]"),RadioPara.ActiveName);
+        _stprintf(Name,_T("[%s]"),ActiveName);
     if(wpnewActive)
       wpnewActive->SetCaption(Name);
     _stprintf(Name,_T("%6.03f"),RadioPara.ActiveFrequency);
     if(wpnewActiveFreq)
       wpnewActiveFreq->SetCaption(Name);
 
 
-    if(_tcscmp(_T("        "), RadioPara.PassiveName ) == 0)
-    Idx = SearchStation(RadioPara.PassiveFrequency);
+    if((_tcscmp(_T("        "), RadioPara.PassiveName ) == 0) ||( _tcslen(RadioPara.PassiveName) == 0))
```

Comment:
same as line 160

Reply:
Some radios (KRT2) send an string full of spaces others an empty string

Reply:
in this case name must be trimmed when it is received ...